src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryUnary.cs

```diff
@@ -454,6 +456,66 @@ public static FormulaValue OptionSetValueToString(IRContext irContext, OptionSet
             var displayName = optionSet.DisplayName;
             return new StringValue(irContext, displayName);
         }
+
+        public static FormulaValue TextToColor(IRContext irContext, StringValue[] args)
+        {
+            var val = args[0].Value;
+
+            if (string.IsNullOrEmpty(val))
+            {
+                return new BlankValue(irContext);
+            }
+
+            var regex = new Regex(@"^#(?<r>[0-9a-fA-F]{2})(?<g>[0-9a-fA-F]{2})(?<b>[0-9a-fA-F]{2})(?<a>[0-9a-fA-F]{2})?$");
+            var match = regex.Match(val);
+            if (match.Success)
+            {
+                var r = byte.Parse(match.Groups["r"].Value, System.Globalization.NumberStyles.HexNumber);
+                var g = byte.Parse(match.Groups["g"].Value, System.Globalization.NumberStyles.HexNumber);
+                var b = byte.Parse(match.Groups["b"].Value, System.Globalization.NumberStyles.HexNumber);
+                var a = match.Groups["a"].Captures.Count == 1 ? byte.Parse(match.Groups["a"].Value, System.Globalization.NumberStyles.HexNumber) : 255;
+
+                return new ColorValue(irContext, Color.FromArgb(a, r, g, b));
+            }
+
+            return CommonErrors.InvalidColorFormatError(irContext);
+        }
+
+        public static FormulaValue RGBA(IRContext irContext, NumberValue[] args)
+        {
+            if (args.Length != 4)
+            {
+                return CommonErrors.GenericInvalidArgument(irContext);
+            }
+
+            // Ensure rgb numbers are in range (0-255)
+            if (args[0].Value < 0.0d || args[0].Value > 255.0d
+                || args[1].Value < 0.0d || args[1].Value > 255.0d
+                || args[2].Value < 0.0d || args[2].Value > 255.0d)
+            {
+                return CommonErrors.ArgumentOutOfRange(irContext);
+            }
+
+            var r = System.Convert.ToByte(args[0].Value);
+            var g = System.Convert.ToByte(args[1].Value);
+            var b = System.Convert.ToByte(args[2].Value);
+
+            // Ensure bytes are whole numbers and weren't rounded
+            if (r != args[0].Value || g != args[1].Value || b != args[2].Value)
+            {
+                return CommonErrors.InvalidNumberFormatError(irContext);
+            }
+
+            // Ensure alpha is between 0 and 1
+            if (args[3].Value < 0.0d || args[3].Value > 1.0d)
+            {
+                return CommonErrors.ArgumentOutOfRange(irContext);
+            }
+
+            var a = System.Convert.ToInt32(args[3].Value * 255.0d);
+
+            return new ColorValue(irContext, Color.FromArgb(a, r, g, b));
+        }
         #endregion
```

Comment:
These should probably go to a new file (LibraryColor.cs) - they are not unary operators / implicit coercions.

Reply:
Yes, not sure why I started in there! Done.